Runtimes/Unity/BepInEx.Unity.IL2CPP/Il2CppInteropManager.cs

```diff
@@ -282,25 +283,24 @@ private static void GenerateInteropAssemblies()
         }
     }
 
-    private static void DownloadUnityAssemblies()
-    {
+    private static void DownloadUnityAssemblies() {
         var unityVersion = UnityInfo.Version;
-        var source =
-            UnityBaseLibrariesSource.Value.Replace("{VERSION}",
-                                                   $"{unityVersion.Major}.{unityVersion.Minor}.{unityVersion.Build}");
-
-        if (!string.IsNullOrEmpty(source))
-        {
-            Logger.LogMessage("Downloading unity base libraries");
-
-            Directory.CreateDirectory(UnityBaseLibsDirectory);
-            Directory.EnumerateFiles(UnityBaseLibsDirectory, "*.dll").Do(File.Delete);
-
+        var file = $"{unityVersion.Major}.{unityVersion.Minor}.{unityVersion.Build}.zip";
+
+        var baseFolder = Directory.CreateDirectory(UnityBaseLibsDirectory);
+        baseFolder.EnumerateFiles("*.dll").Do(a=>a.Delete());
+        var target = baseFolder.GetFiles(file).FirstOrDefault();
+        if (target != null) {
+            using var fStream = target.OpenRead();
+            using var zipArchive = new ZipArchive(fStream, ZipArchiveMode.Read);
+            zipArchive.ExtractToDirectory(UnityBaseLibsDirectory);
+        } else {
+            var source = UnityBaseLibrariesSource.Value.Replace("{VERSION}.zip", file);
```

Comment:
Requiring the source setting to always add a zip at the end of `{VERSION}.zip` is a breaking change.

Reply:
well, System.IO.Compression.ZipArchive must be always a zip file but I'll revert this change. It just makes it easier to get the local file

Reply:
It could be `{VERSION}_blah.zip` etc.